synphot/observation.py

```diff
@@ -694,3 +694,220 @@ def as_spectrum(self, binned=True, wavelengths=None):
         header = {'observation': str(self), 'binned': binned}
         return SourceSpectrum(Empirical1D, points=w, lookup_table=y,
                               meta={'header': header})
+
+
+# See ad_err documentation below.
+AD_ERR_DEFAULT = np.sqrt(0.289) * (u.adu / u.pixel)
+
+
+@u.quantity_input(counts=u.electron,
+                  npix=u.pixel,
+                  n_background=u.pixel,
+                  background=u.electron / u.pixel,
+                  darkcurrent=u.electron / u.pixel,
+                  readnoise=u.electron / u.pixel,
+                  gain=u.electron / u.adu,
+                  ad_err=u.adu / u.pixel)
+def howell_snr(counts,
+               npix=1 * u.pixel,
+               n_background=np.inf * u.pixel,
+               background=0 * (u.electron / u.pixel),
+               darkcurrent=0 * (u.electron / u.pixel),
+               readnoise=0 * (u.electron / u.pixel),
+               gain=1 * (u.electron / u.adu),
+               ad_err=AD_ERR_DEFAULT):
+    """
+    A function to calculate the idealized theoretical signal to noise ratio
+    (SNR) of an astronomical observation with a given number of counts. This
+    expression is taken from pg 55 of [1]_).
+
+    Parameters
+    ----------
+    counts : `~astropy.units.Quantity`
```

Comment:
This *might* be a matter of me missing a "higher" layer, but I think the more useful input here is *flux*.  That is, most astronomers don't know the number of counts, but they do know the flux of their target (or in the given spectral pixel, or whatever).

Or is the intent to have this be a "lower layer" that is called by other functions that just translate flux to counts?

Reply:
I might be misunderstanding the question but I think we use counts here because we only care about the number of photons hitting the ccd, not the total energy absorbed by the ccd (which is what I think of when I think of "flux", i.e. W/m^2). I do see how counts can be labeled as flux if you think of flux as just the "net amount of stuff going through an area", but I would easily confuse the parameter to mean "power density" if we were to change it.

Reply:
I guess a higher layer can be a user-facing example that sets up the Observation first, then calculates the count rate, and then pass the count rate into this function?

Reply:
> I do see how counts can be labeled as flux if you think of flux as just the

Ah sorry, not quite what I meant here - rather I was thinking in the direction @pllim is saying, and I skipped a few steps.  That is, I agree we *don't* want to do the conversion you're saying here, @tcjansen, but rather what @pllim suggested, that this could be the "last" function in a set of calls.

In the end, this might end up being one of the "high-level" functions @tcjansen might adapt from some of the notebooks, I think. At some point it might be desirable to note that in the docs here, but I guess it's not good to do that until the higher level function exists to point the reader to.